File: cpp-tiny/lib/Models/BasicStreamAllOf.cpp


#include "BasicStream_allOf.h"

using namespace Tiny;

BasicStream_allOf::BasicStream_allOf()
{
	stream_id = null;
	name = null;
	description = null;
	date_created = null;
	invite_only = null;
	rendered_description = null;
	is_web_public = null;
	stream_post_policy = null;
	message_retention_days = null;
	history_public_to_subscribers = null;
	first_message_id = null;
	is_announcement_only = null;
}

BasicStream_allOf::BasicStream_allOf(std::string jsonString)
{
	this->fromJson(jsonString);
}

BasicStream_allOf::~BasicStream_allOf()
{

}

void
BasicStream_allOf::fromJson(std::string jsonObj)
{   
    bourne::json object = bourne::json::parse(jsonObj);

    const char *stream_idKey = "stream_id";
    
    if(object.has_key(stream_idKey)) 
    {
        bourne::json value = object[stream_idKey];


        

        AnyType* obj = &stream_id;
		obj->fromJson(value.dump());

    }

    const char *nameKey = "name";
    
    if(object.has_key(nameKey)) 
    {
        bourne::json value = object[nameKey];


        

        AnyType* obj = &name;
		obj->fromJson(value.dump());

    }

    const char *descriptionKey = "description";
    
    if(object.has_key(descriptionKey)) 
    {
        bourne::json value = object[descriptionKey];


        

        AnyType* obj = &description;
		obj->fromJson(value.dump());

    }

    const char *date_createdKey = "date_created";
    
    if(object.has_key(date_createdKey)) 
    {
        bourne::json value = object[date_createdKey];


        

        AnyType* obj = &date_created;
		obj->fromJson(value.dump());

    }

    const char *invite_onlyKey = "invite_only";
    
    if(object.has_key(invite_onlyKey)) 
    {
        bourne::json value = object[invite_onlyKey];


        

        AnyType* obj = &invite_only;
		obj->fromJson(value.dump());

    }

    const char *rendered_descriptionKey = "rendered_description";
    
    if(object.has_key(rendered_descriptionKey)) 
    {
        bourne::json value = object[rendered_descriptionKey];


        

        AnyType* obj = &rendered_description;
		obj->fromJson(value.dump());

    }

    const char *is_web_publicKey = "is_web_public";
    
    if(object.has_key(is_web_publicKey)) 
    {
        bourne::json value = object[is_web_publicKey];


        

        AnyType* obj = &is_web_public;
		obj->fromJson(value.dump());

    }

    const char *stream_post_policyKey = "stream_post_policy";
    
    if(object.has_key(stream_post_policyKey)) 
    {
        bourne::json value = object[stream_post_policyKey];


        

        AnyType* obj = &stream_post_policy;
		obj->fromJson(value.dump());

    }

    const char *message_retention_daysKey = "message_retention_days";
    
    if(object.has_key(message_retention_daysKey)) 
    {
        bourne::json value = object[message_retention_daysKey];


        

        AnyType* obj = &message_retention_days;
		obj->fromJson(value.dump());

    }

    const char *history_public_to_subscribersKey = "history_public_to_subscribers";
    
    if(object.has_key(history_public_to_subscribersKey)) 
    {
        bourne::json value = object[history_public_to_subscribersKey];


        

        AnyType* obj = &history_public_to_subscribers;
		obj->fromJson(value.dump());

    }

    const char *first_message_idKey = "first_message_id";
    
    if(object.has_key(first_message_idKey)) 
    {
        bourne::json value = object[first_message_idKey];


        

        AnyType* obj = &first_message_id;
		obj->fromJson(value.dump());

    }

    const char *is_announcement_onlyKey = "is_announcement_only";
    
    if(object.has_key(is_announcement_onlyKey)) 
    {
        bourne::json value = object[is_announcement_onlyKey];


        

        AnyType* obj = &is_announcement_only;
		obj->fromJson(value.dump());

    }


}

bourne::json
BasicStream_allOf::toJson()
{
    bourne::json object = bourne::json::object();

    




	object["stream_id"] = getStreamId().toJson();

    




	object["name"] = getName().toJson();

    




	object["description"] = getDescription().toJson();

    




	object["date_created"] = getDateCreated().toJson();

    




	object["invite_only"] = getInviteOnly().toJson();

    




	object["rendered_description"] = getRenderedDescription().toJson();

    




	object["is_web_public"] = getIsWebPublic().toJson();

    




	object["stream_post_policy"] = getStreamPostPolicy().toJson();

    




	object["message_retention_days"] = getMessageRetentionDays().toJson();

    




	object["history_public_to_subscribers"] = getHistoryPublicToSubscribers().toJson();

    




	object["first_message_id"] = getFirstMessageId().toJson();

    




	object["is_announcement_only"] = getIsAnnouncementOnly().toJson();


    return object;

}

AnyType
BasicStream_allOf::getStreamId()
{
	return stream_id;
}

void
BasicStream_allOf::setStreamId(AnyType  stream_id)
{
	this->stream_id = stream_id;
}

AnyType
BasicStream_allOf::getName()
{
	return name;
}

void
BasicStream_allOf::setName(AnyType  name)
{
	this->name = name;
}

AnyType
BasicStream_allOf::getDescription()
{
	return description;
}

void
BasicStream_allOf::setDescription(AnyType  description)
{
	this->description = description;
}

AnyType
BasicStream_allOf::getDateCreated()
{
	return date_created;
}

void
BasicStream_allOf::setDateCreated(AnyType  date_created)
{
	this->date_created = date_created;
}

AnyType
BasicStream_allOf::getInviteOnly()
{
	return invite_only;
}

void
BasicStream_allOf::setInviteOnly(AnyType  invite_only)
{
	this->invite_only = invite_only;
}

AnyType
BasicStream_allOf::getRenderedDescription()
{
	return rendered_description;
}

void
BasicStream_allOf::setRenderedDescription(AnyType  rendered_description)
{
	this->rendered_description = rendered_description;
}

AnyType
BasicStream_allOf::getIsWebPublic()
{
	return is_web_public;
}

void
BasicStream_allOf::setIsWebPublic(AnyType  is_web_public)
{
	this->is_web_public = is_web_public;
}

AnyType
BasicStream_allOf::getStreamPostPolicy()
{
	return stream_post_policy;
}

void
BasicStream_allOf::setStreamPostPolicy(AnyType  stream_post_policy)
{
	this->stream_post_policy = stream_post_policy;
}

AnyType
BasicStream_allOf::getMessageRetentionDays()
{
	return message_retention_days;
}

void
BasicStream_allOf::setMessageRetentionDays(AnyType  message_retention_days)
{
	this->message_retention_days = message_retention_days;
}

AnyType
BasicStream_allOf::getHistoryPublicToSubscribers()
{
	return history_public_to_subscribers;
}

void
BasicStream_allOf::setHistoryPublicToSubscribers(AnyType  history_public_to_subscribers)
{
	this->history_public_to_subscribers = history_public_to_subscribers;
}

AnyType
BasicStream_allOf::getFirstMessageId()
{
	return first_message_id;
}

void
BasicStream_allOf::setFirstMessageId(AnyType  first_message_id)
{
	this->first_message_id = first_message_id;
}

AnyType
BasicStream_allOf::getIsAnnouncementOnly()
{
	return is_announcement_only;
}

void
BasicStream_allOf::setIsAnnouncementOnly(AnyType  is_announcement_only)
{
	this->is_announcement_only = is_announcement_only;
}

File: spring/src/main/java/org/openapitools/model/BasicStreamAllOf.java
package org.openapitools.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
import org.openapitools.jackson.nullable.JsonNullable;
import javax.validation.Valid;
import javax.validation.constraints.*;

/**
 * BasicStreamAllOf
 */
@javax.annotation.Generated(value = "org.openapitools.codegen.languages.SpringCodegen")
public class BasicStreamAllOf   {
  @JsonProperty("stream_id")
  private JsonNullable<Object> streamId = JsonNullable.undefined();

  @JsonProperty("name")
  private JsonNullable<Object> name = JsonNullable.undefined();

  @JsonProperty("description")
  private JsonNullable<Object> description = JsonNullable.undefined();

  @JsonProperty("date_created")
  private JsonNullable<Object> dateCreated = JsonNullable.undefined();

  @JsonProperty("invite_only")
  private JsonNullable<Object> inviteOnly = JsonNullable.undefined();

  @JsonProperty("rendered_description")
  private JsonNullable<Object> renderedDescription = JsonNullable.undefined();

  @JsonProperty("is_web_public")
  private JsonNullable<Object> isWebPublic = JsonNullable.undefined();

  @JsonProperty("stream_post_policy")
  private JsonNullable<Object> streamPostPolicy = JsonNullable.undefined();

  @JsonProperty("message_retention_days")
  private JsonNullable<Object> messageRetentionDays = JsonNullable.undefined();

  @JsonProperty("history_public_to_subscribers")
  private JsonNullable<Object> historyPublicToSubscribers = JsonNullable.undefined();

  @JsonProperty("first_message_id")
  private JsonNullable<Object> firstMessageId = JsonNullable.undefined();

  @JsonProperty("is_announcement_only")
  private JsonNullable<Object> isAnnouncementOnly = JsonNullable.undefined();

  public BasicStreamAllOf streamId(Object streamId) {
    this.streamId = JsonNullable.of(streamId);
    return this;
  }

  /**
   * Get streamId
   * @return streamId
  */
  @ApiModelProperty(value = "")


  public JsonNullable<Object> getStreamId() {
    return streamId;
  }

  public void setStreamId(JsonNullable<Object> streamId) {
    this.streamId = streamId;
  }

  public BasicStreamAllOf name(Object name) {
    this.name = JsonNullable.of(name);
    return this;
  }

  /**
   * Get name
   * @return name
  */
  @ApiModelProperty(value = "")


  public JsonNullable<Object> getName() {
    return name;
  }

  public void setName(JsonNullable<Object> name) {
    this.name = name;
  }

  public BasicStreamAllOf description(Object description) {
    this.description = JsonNullable.of(description);
    return this;
  }

  /**
   * Get description
   * @return description
  */
  @ApiModelProperty(value = "")


  public JsonNullable<Object> getDescription() {
    return description;
  }

  public void setDescription(JsonNullable<Object> description) {
    this.description = description;
  }

  public BasicStreamAllOf dateCreated(Object dateCreated) {
    this.dateCreated = JsonNullable.of(dateCreated);
    return this;
  }

  /**
   * Get dateCreated
   * @return dateCreated
  */
  @ApiModelProperty(value = "")


  public JsonNullable<Object> getDateCreated() {
    return dateCreated;
  }

  public void setDateCreated(JsonNullable<Object> dateCreated) {
    this.dateCreated = dateCreated;
  }

  public BasicStreamAllOf inviteOnly(Object inviteOnly) {
    this.inviteOnly = JsonNullable.of(inviteOnly);
    return this;
  }

  /**
   * Get inviteOnly
   * @return inviteOnly
  */
  @ApiModelProperty(value = "")


  public JsonNullable<Object> getInviteOnly() {
    return inviteOnly;
  }

  public void setInviteOnly(JsonNullable<Object> inviteOnly) {
    this.inviteOnly = inviteOnly;
  }

  public BasicStreamAllOf renderedDescription(Object renderedDescription) {
    this.renderedDescription = JsonNullable.of(renderedDescription);
    return this;
  }

  /**
   * Get renderedDescription
   * @return renderedDescription
  */
  @ApiModelProperty(value = "")


  public JsonNullable<Object> getRenderedDescription() {
    return renderedDescription;
  }

  public void setRenderedDescription(JsonNullable<Object> renderedDescription) {
    this.renderedDescription = renderedDescription;
  }

  public BasicStreamAllOf isWebPublic(Object isWebPublic) {
    this.isWebPublic = JsonNullable.of(isWebPublic);
    return this;
  }

  /**
   * Get isWebPublic
   * @return isWebPublic
  */
  @ApiModelProperty(value = "")


  public JsonNullable<Object> getIsWebPublic() {
    return isWebPublic;
  }

  public void setIsWebPublic(JsonNullable<Object> isWebPublic) {
    this.isWebPublic = isWebPublic;
  }

  public BasicStreamAllOf streamPostPolicy(Object streamPostPolicy) {
    this.streamPostPolicy = JsonNullable.of(streamPostPolicy);
    return this;
  }

  /**
   * Get streamPostPolicy
   * @return streamPostPolicy
  */
  @ApiModelProperty(value = "")


  public JsonNullable<Object> getStreamPostPolicy() {
    return streamPostPolicy;
  }

  public void setStreamPostPolicy(JsonNullable<Object> streamPostPolicy) {
    this.streamPostPolicy = streamPostPolicy;
  }

  public BasicStreamAllOf messageRetentionDays(Object messageRetentionDays) {
    this.messageRetentionDays = JsonNullable.of(messageRetentionDays);
    return this;
  }

  /**
   * Get messageRetentionDays
   * @return messageRetentionDays
  */
  @ApiModelProperty(value = "")


  public JsonNullable<Object> getMessageRetentionDays() {
    return messageRetentionDays;
  }

  public void setMessageRetentionDays(JsonNullable<Object> messageRetentionDays) {
    this.messageRetentionDays = messageRetentionDays;
  }

  public BasicStreamAllOf historyPublicToSubscribers(Object historyPublicToSubscribers) {
    this.historyPublicToSubscribers = JsonNullable.of(historyPublicToSubscribers);
    return this;
  }

  /**
   * Get historyPublicToSubscribers
   * @return historyPublicToSubscribers
  */
  @ApiModelProperty(value = "")


  public JsonNullable<Object> getHistoryPublicToSubscribers() {
    return historyPublicToSubscribers;
  }

  public void setHistoryPublicToSubscribers(JsonNullable<Object> historyPublicToSubscribers) {
    this.historyPublicToSubscribers = historyPublicToSubscribers;
  }

  public BasicStreamAllOf firstMessageId(Object firstMessageId) {
    this.firstMessageId = JsonNullable.of(firstMessageId);
    return this;
  }

  /**
   * Get firstMessageId
   * @return firstMessageId
  */
  @ApiModelProperty(value = "")


  public JsonNullable<Object> getFirstMessageId() {
    return firstMessageId;
  }

  public void setFirstMessageId(JsonNullable<Object> firstMessageId) {
    this.firstMessageId = firstMessageId;
  }

  public BasicStreamAllOf isAnnouncementOnly(Object isAnnouncementOnly) {
    this.isAnnouncementOnly = JsonNullable.of(isAnnouncementOnly);
    return this;
  }

  /**
   * Get isAnnouncementOnly
   * @return isAnnouncementOnly
  */
  @ApiModelProperty(value = "")


  public JsonNullable<Object> getIsAnnouncementOnly() {
    return isAnnouncementOnly;
  }

  public void setIsAnnouncementOnly(JsonNullable<Object> isAnnouncementOnly) {
    this.isAnnouncementOnly = isAnnouncementOnly;
  }


  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    BasicStreamAllOf basicStreamAllOf = (BasicStreamAllOf) o;
    return Objects.equals(this.streamId, basicStreamAllOf.streamId) &&
        Objects.equals(this.name, basicStreamAllOf.name) &&
        Objects.equals(this.description, basicStreamAllOf.description) &&
        Objects.equals(this.dateCreated, basicStreamAllOf.dateCreated) &&
        Objects.equals(this.inviteOnly, basicStreamAllOf.inviteOnly) &&
        Objects.equals(this.renderedDescription, basicStreamAllOf.renderedDescription) &&
        Objects.equals(this.isWebPublic, basicStreamAllOf.isWebPublic) &&
        Objects.equals(this.streamPostPolicy, basicStreamAllOf.streamPostPolicy) &&
        Objects.equals(this.messageRetentionDays, basicStreamAllOf.messageRetentionDays) &&
        Objects.equals(this.historyPublicToSubscribers, basicStreamAllOf.historyPublicToSubscribers) &&
        Objects.equals(this.firstMessageId, basicStreamAllOf.firstMessageId) &&
        Objects.equals(this.isAnnouncementOnly, basicStreamAllOf.isAnnouncementOnly);
  }

  @Override
  public int hashCode() {
    return Objects.hash(streamId, name, description, dateCreated, inviteOnly, renderedDescription, isWebPublic, streamPostPolicy, messageRetentionDays, historyPublicToSubscribers, firstMessageId, isAnnouncementOnly);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class BasicStreamAllOf {\n");
    
    sb.append("    streamId: ").append(toIndentedString(streamId)).append("\n");
    sb.append("    name: ").append(toIndentedString(name)).append("\n");
    sb.append("    description: ").append(toIndentedString(description)).append("\n");
    sb.append("    dateCreated: ").append(toIndentedString(dateCreated)).append("\n");
    sb.append("    inviteOnly: ").append(toIndentedString(inviteOnly)).append("\n");
    sb.append("    renderedDescription: ").append(toIndentedString(renderedDescription)).append("\n");
    sb.append("    isWebPublic: ").append(toIndentedString(isWebPublic)).append("\n");
    sb.append("    streamPostPolicy: ").append(toIndentedString(streamPostPolicy)).append("\n");
    sb.append("    messageRetentionDays: ").append(toIndentedString(messageRetentionDays)).append("\n");
    sb.append("    historyPublicToSubscribers: ").append(toIndentedString(historyPublicToSubscribers)).append("\n");
    sb.append("    firstMessageId: ").append(toIndentedString(firstMessageId)).append("\n");
    sb.append("    isAnnouncementOnly: ").append(toIndentedString(isAnnouncementOnly)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}

